File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/setting/SettingDao.java
package ru.nikshlykov.englishwordsapp.db.setting;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface SettingDao {

    @Update
    int update(Setting setting);

    @Update
    int update(Setting[] settings);

    // Получаем все настройки.
    @Query("SELECT * FROM Settings")
    Setting[] getAllSettings();
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/setting/SettingDao.kt
package ru.nikshlykov.englishwordsapp.db.setting

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import androidx.room.Update

@Dao
interface SettingDao {
  @Update
  fun update(setting: Setting?): Int

  @Update
  fun update(settings: Array<Setting?>?): Int

  // Получаем все настройки.
  @Query("SELECT * FROM Settings")
  fun allSettings(): LiveData<List<Setting?>?>?
}